editor/filesystem_dock.cpp

```diff
@@ -1923,24 +1928,27 @@ void FileSystemDock::_file_option(int p_option, const Vector<String> &p_selected
 		} break;
 
 		case FILE_RENAME: {
-			// Rename the active file.
 			if (!p_selected.is_empty()) {
 				to_rename.path = p_selected[0];
-				if (to_rename.path != "res://") {
-					to_rename.is_file = !to_rename.path.ends_with("/");
-					if (to_rename.is_file) {
-						String name = to_rename.path.get_file();
-						rename_dialog->set_title(TTR("Renaming file:") + " " + name);
-						rename_dialog_text->set_text(name);
-						rename_dialog_text->select(0, name.rfind("."));
-					} else {
-						String name = to_rename.path.substr(0, to_rename.path.length() - 1).get_file();
-						rename_dialog->set_title(TTR("Renaming folder:") + " " + name);
-						rename_dialog_text->set_text(name);
-						rename_dialog_text->select(0, name.length());
-					}
-					rename_dialog->popup_centered(Size2(250, 80) * EDSCALE);
-					rename_dialog_text->grab_focus();
+			}
+
+			if (tree->is_anything_selected() && to_rename.path != "res://") {
+				// Edit node in Tree
+				tree->grab_focus();
+				TreeItem *s = tree->get_selected();
+				s->set_editable(tree->get_selected_column(), true);
```

Comment:
I think the TreeItems can be made editable when they are created (inside `_create_tree()`). Note that there are exceptions, namely `res://` and `Favorites:`.

Reply:
Yep, the "Favorites:" node is currently editable which is undesirable.
I'll check for is_anything_selected(). I based my edits on the SceneTree implementation.

I'll make a second revision tonight.